src/home_robot/home_robot/utils/point_cloud_torch.py

```diff
@@ -0,0 +1,143 @@
+"""
+This file contains versions of the helpers in point_cloud.py that use pytorch directly (rather than numpy),
+to allow operations to be done on the GPU for speed.
+"""
+import cv2
+import numpy as np
+import torch
+from torch_geometric.nn.pool.voxel_grid import voxel_grid
+from home_robot.utils.image import Camera
+
+
+def depth_to_xyz(depth, camera: Camera):
+    """get depth from numpy using simple pinhole camera model"""
+    indices = np.indices((camera.height, camera.width), dtype=np.float32).transpose(1, 2, 0)
+    z = depth
+
+    # pixel indices start at top-left corner. for these equations, it starts at bottom-left
+    x = torch.tensor(indices[:, :, 1] - camera.px).to(z.device) * (z / camera.fx)
+    y = torch.tensor(indices[:, :, 0] - camera.py).to(z.device) * (z / camera.fy)
+
+    # Should now be batch x height x width x 3, after this:
+    xyz = torch.stack([x, y, z], axis=-1)
+    return xyz
+
+
+def add_additive_noise_to_xyz(
```

Comment:
These functions are all slight tweaks of the existing functions. I considered just handling the torch case directly in the existing functionality, but it got a little unwieldy, and at least this way we can optimize further if the need strikes. (I didn't switch everything to torch, just the operations on the big arrays.) Can switch to the other way if desired -- not a huge fan of the code duplication, but this way was less intrusive to existing code.